### ISSUE TITLE
pass_placeholder flags abstract methods with pass body

### ISSUE THREAD

**Issue by rsionnach** [rsionnach/sloppylint#5]
## Issue
The `pass_placeholder` pattern incorrectly flags abstract methods that use `pass` as their body.

## Example
```python
from abc import ABC, abstractmethod

class MyBase(ABC):
    @abstractmethod
    def required_method(self):
        pass  # FLAGGED - but this is valid!
```

## Expected Behavior
Methods decorated with `@abstractmethod` (or inside Protocol/ABC classes) should not be flagged when using `pass` as the body, as this is a standard Python pattern for abstract method definitions.

## Suggested Fix
Apply the same fix as `ellipsis_placeholder` - check for abstract decorators and Protocol/ABC class context.

**Comment by rsionnach** [comment]
This was already addressed. The `pass_placeholder` pattern checks for:

1. **Abstract decorators**: `@abstractmethod`, `@abstractproperty`, `@abstractclassmethod`, `@abstractstaticmethod`, `@overload`
2. **Protocol/ABC context**: Methods inside classes inheriting from `Protocol`, `ABC`, `ABCMeta`, `Interface`, `typing_extensions`

Methods with these decorators or in these contexts are NOT flagged when using `pass` as the body.

Closing as fixed.
